src/MQTTManager.cpp

```diff
@@ -114,205 +121,154 @@ void MQTTMessageDispatcher::setRetainFlag(int retainFlag) {
                                               esp_event_base_t base,
                                               int32_t event_id,
                                               void* event_data) {
-    MQTTMessageDispatcher* mqttMgr =
-        reinterpret_cast<MQTTMessageDispatcher*>(handler_args);
-
-    ESP_LOGD(mqttMgr->TAG,
-             "Event dispatched from event loop base=%s, event_id=%" PRIi32 "",
-             base, event_id);
-    esp_mqtt_event_handle_t event =
-        reinterpret_cast<esp_mqtt_event_handle_t>(event_data);
-    esp_mqtt_client_handle_t client = event->client;
+    auto* mqttMgr = reinterpret_cast<MqttMailingService*>(handler_args);
+    auto event = reinterpret_cast<esp_mqtt_event_handle_t>(event_data);
 
     switch ((esp_mqtt_event_id_t)event_id) {
         case MQTT_EVENT_CONNECTED:
-            ESP_LOGI(mqttMgr->TAG, "MQTT_EVENT_CONNECTED");
+            ESP_LOGI(mqttMgr->TAG, "ESP MQTT client connected");
+            mqttMgr->_state = MqttMailingServiceState::CONNECTED;
+
+            // Start running task
             if (mqttMgr->_mailmanTaskHandle == nullptr) {
-                xTaskCreate(mqttMgr->_mailmanTask, "MQTT MsgDispatcher",
-                            5 * 1024, reinterpret_cast<void*>(mqttMgr),
-                            tskIDLE_PRIORITY + 1,
-                            &(mqttMgr->_mailmanTaskHandle));
+                xTaskCreate(
+                    MqttMailingService::_mailmanTask, "MQTT MsgDispatcher",
+                    5 * 1024, reinterpret_cast<void*>(mqttMgr),
+                    tskIDLE_PRIORITY + 1, &(mqttMgr->_mailmanTaskHandle));
                 ESP_LOGI(mqttMgr->TAG, "Mailman task launched.");
             }
-            mqttMgr->_clientState = MQTT_STATE_CONNECTED;
-            mqttMgr->_mailmanIsRunning = true;
             break;
         case MQTT_EVENT_DISCONNECTED:
-            ESP_LOGI(mqttMgr->TAG, "MQTT_EVENT_DISCONNECTED");
-            mqttMgr->_mailmanIsRunning = false;
-            mqttMgr->_clientState = MQTT_STATE_DISCONNECTED;
-            xTaskAbortDelay(mqttMgr->_mailmanTaskHandle);
+            ESP_LOGI(mqttMgr->TAG, "ESP MQTT client disconnected");
+            mqttMgr->_state = MqttMailingServiceState::DISCONNECTED;
             break;
         case MQTT_EVENT_ERROR:
-            ESP_LOGW(mqttMgr->TAG, "MQTT_EVENT_ERROR");
+            ESP_LOGW(mqttMgr->TAG,
+                     "ESP MQTT client encountered an error (code %i)",
+                     (*event).error_handle->error_type);
             break;
         default:
             break;
     }
 }
+
+/**
+ * Wi-Fi check task
+ * This tasks checks if Wi-Fi is still connected once per second
+ * and tries to reconnect if Wi-Fi connection is not established.
+ */
+void MqttMailingService::_wifiCheckTask(__attribute__((unused)) void* arg) {
+    while (true) {
+        vTaskDelay(pdMS_TO_TICKS(WIFI_CHECK_INTERVAL_MS));
+
+        if (WiFiClass::status() != WL_CONNECTED) {
```

Comment:
```suggestion
        if (!WiFiClass::isConnected()) {
```

Reply:
status is static so we don't have to access the WiFi object. I prefer it here, since this is running in a separate thread

Reply:
ok, makes sense